### PR TITLE
Updated pull request template (for documentation)

### PR DESCRIPTION
**Pull request by evildmp**
[skip-ci]

### Summary

Aims to improve documentation processes, by providing a checklist of actions.

In particular, this should make release notes more accurate, complete and easier to manage.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -11,3 +16,18 @@ Fixes #
...
+### Documentation checklist
+
+* [ ] I have added a line to CHANGELOG.txt
+* [ ] I have updated the release notes document
+    * [ ] general notes
+    * [ ] bug-fixes
+    * [ ] improvements/new features
+    * [ ] backwards-imcompatible changes
+    * [ ] required upgrade steps
+    * [ ] names of community contributors
```
Comment (czpython) [inline .github/PULL_REQUEST_TEMPLATE.md]: what would be an example of this?

Comment (evildmp) [reply .github/PULL_REQUEST_TEMPLATE.md]: Actually I think it should just be all contributors with commits in that particular release, in the form
 ```
* Daniele Procida <http://github.com/evildmp/>
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -11,3 +16,18 @@ Fixes #
 
 
 ### Proposed changes in this pull request
+
+
+### Documentation checklist
+
+* [ ] I have added a line to CHANGELOG.txt
```
Comment (evildmp) [inline .github/PULL_REQUEST_TEMPLATE.md]: or multiple lines - and only for non-regressions.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,6 +1,11 @@
-### Summary
+<!--
+    If this is a security-related patch stop immediately!
+
+    See http://docs.django-cms.org/en/release-3.4.x/contributing/development-policies.html
```
Comment (czpython) [inline .github/PULL_REQUEST_TEMPLATE.md]: maybe use "latest" here instead of 3.4.x

**Comment by FinalAngel** [comment]
@evildmp @czpython how do we stand here?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -11,3 +16,18 @@ Fixes #
 
 
 ### Proposed changes in this pull request
+
+
+### Documentation checklist
+
+* [ ] I have updated CHANGELOG.txt if appropriate
+* [ ] I have updated the release notes document if appropriate, with:
```
Comment (FinalAngel) [inline .github/PULL_REQUEST_TEMPLATE.md]: maybe add links to them?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -11,3 +16,18 @@ Fixes #
...
+    * [ ] general notes
+    * [ ] bug-fixes
+    * [ ] improvements/new features
+    * [ ] backwards-incompatible changes
+    * [ ] required upgrade steps
+    * [ ] names of contributors
+* [ ] I have updated other documentation
```
Comment (FinalAngel) [inline .github/PULL_REQUEST_TEMPLATE.md]: Be specific?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -11,3 +16,18 @@ Fixes #
...
+* [ ] I have added my name to the AUTHORS file
+* [ ] This PR's documentation has been approved by Daniele Procida
```
Comment (FinalAngel) [inline .github/PULL_REQUEST_TEMPLATE.md]: Why you add your name here? not something like your.name?

**Pull request merged**